### ISSUE TITLE
[FEATURE] Validate NET_RAW probes, possible error is suspected

### ISSUE THREAD

**Issue by lcarrazana** [citihub/probr#259]
**Describe the bug**
The test for this scenario is not consistent and must be reviewed/updated.

@k-psp-007
Scenario Outline: Prevent deployments from running with the NET_RAW capability.

**To Reproduce**
1. Run probr with --tags=@k-psp-007
2. Error: TBD

**Expected behavior**
TBD

**Additional Context**
This is an extension of #95

**Comment by lcarrazana** [comment]
```
@k-psp-007
    Scenario Outline: Prevent deployments from running with the NET_RAW capability.
        Given a Kubernetes cluster exists which we can deploy into
        And some system exists to prevent a Kubernetes deployment from running with NET_RAW capability in an existing Kubernetes cluster
        When a Kubernetes deployment is applied to an existing Kubernetes cluster
        And  NET_RAW is marked "<NET_RAW requested>" for the Kubernetes deployment
        Then the operation will "<RESULT>" with an error "<ERRORMESSAGE>"
        But I should not be able to perform a command that requires NET_RAW capability
        And I should be able to perform an allowed command
```

**Comment by lcarrazana** [comment]
When running `@k-psp-007` against demo cluster, the test fails on step 2 because no Pod Security Policy exists in Kube cluster. This is expected behavior for test. The Azure policy is not recognized because it is not handled in code
Potential solutions for passing step 2:
1. Deploy at least one PSP with NEW_RAW capbality drop to K8s cluster (this is best practice as per CIS standard)
2. Update logic inside probe function to check for Azure policies

**Comment by iantivey** [comment]
I vote for stripping this line out of all the PSP scenarios...

`And some system exists to prevent a Kubernetes deployment from running with NET_RAW capability in an existing Kubernetes cluster`

**Comment by eddie-knight** [comment]
relates to #83

**Comment by lcarrazana** [comment]
> I vote for stripping this line out of all the PSP scenarios...
> 
> `And some system exists to prevent a Kubernetes deployment from running with NET_RAW capability in an existing Kubernetes cluster`

As per our discussion on 02/09/21, we are moving ahead with this change request.
As part of this item, I will remove this step for this particular scenario @k-psp-007.
All other scenarios within `pod_security_policy` shall be addressed as part of #83

**Comment by lcarrazana** [comment]
`When a Kubernetes deployment is applied to an existing Kubernetes cluster`

This step is pending for implementation.

**Comment by lcarrazana** [comment]
**Proposed new (and simplified) BDD spec:**

```
@k-psp-007
    Scenario Outline: Prevent deployments from running with the NET_RAW capability.
        Given a Kubernetes cluster exists which we can deploy into
        When a Kubernetes deployment with NET_RAW capability is applied to an existing Kubernetes cluster
        Then the operation will "<RESULT>" with an error "<ERRORMESSAGE>"

        Examples:
            | RESULT  | ERRORMESSAGE                                  |
            | Fail    | Containers cant run with NET_RAW capabilities |
```

**Comment by eddie-knight** [comment]
1. Yes! Remove the "pending implementation" step `aKubernetesDeploymentIsAppliedToAnExistingKubernetesCluster`
2. It's not an unreasonable idea to remove the additional runs for `False` and `Not Defined`. But if we only have one example, we'll then want to adjust this to not be a `Scenario Outline` but simply a `Scenario`. Your most recent comment `Another Proposed BDD spec` has the scenario steps you'd want to use for that.
3. The final two steps are reasonable to leave in, as they provide just a bit more validation to ensure that the policy is being properly applied.

**Comment by lcarrazana** [comment]
**Incorporating @iantivey 's and @eddie-knight 's feedback:**
Assuming cluster is configured to prevent NET_RAW capabilities:
- If a container explicitly adds NET_RAW it should **fail** to deploy
- If a container explicitly drops NET_RAW it should **succeed** to deploy 
- If a container doesn't add neither drop NET_RAW it should **fail** to deploy, since cluster will require explicit drop
```
@k-psp-007
    Scenario Outline: Prevent deployments from running with the NET_RAW capability.
        Given a Kubernetes cluster exists which we can deploy into
        When a Kubernetes deployment with NET_RAW capability <NET_RAW_Option> is applied to an existing Kubernetes cluster
        Then the operation will "<RESULT>" with an error "<ERRORMESSAGE>"

        Examples:
            | NET_RAW_Option | RESULT  | ERRORMESSAGE                                  |
            | Added              | Fail    | Containers cant run with NET_RAW capabilities |
            | Dropped            | Succeed |                                               |
            | Not Defined        | Fail    | Containers cant run with NET_RAW capabilities |
```

---

**Another proposed BDD spec:**
Deploy pod with dropped capability, and prove that executing forbidden commands is not allowed.
Assuming cluster is configured to prevent NET_RAW capabilities:
- A container with explicit drop NET_RAW is deployed
- A command that doesn't require NET_RAW (such as `ls`) should succeed
- A command that requires NET_RAW (such as `ping www.google.com`) should fail
```
@k-psp-007
    Scenario Outline: Prevent deployments from executing commands with the NET_RAW capability.
        Given a Kubernetes cluster exists which we can deploy into
        When a Kubernetes deployment with dropped NET_RAW capability is applied to an existing Kubernetes cluster
        Then the operation will succeed
        And I should be able to perform an allowed command
        But I should not be able to perform a command that requires NET_RAW capability
```

@iantivey What are your thoughts?

**Comment by lcarrazana** [comment]
TODO:

- [ ] When this ticket is closed, retest #270 and verify errors are gone

**Comment by lcarrazana** [comment]
TODO:

- [ ] When executing command (ping google.com) I'm getting this error: `"Error": "runVerificationProbe: {ping google.com 1} - (error in Stream: pod psp-test-pod-110221-234829-64 does not have a host assigned)",` Prob cause is container is in Pending state and has not been scheduled to run on any node yet. May need to add a wait until pod status is Running.
- [ ] There may be an issue with cluster, as pods are not being scheduled
![image](https://user-images.githubusercontent.com/62344673/107733899-1c5d4400-6cca-11eb-998e-27dc52a1e40f.png)
